activerecord/lib/active_record/migration.rb

```diff
@@ -134,7 +134,10 @@ class PendingMigrationError < MigrationError #:nodoc:
     include ActiveSupport::ActionableError
 
     action "Run pending migrations" do
-      ActiveRecord::Tasks::DatabaseTasks.migrate
+      require "rake"
+      Rake::Task.clear
+      Rails.application.load_tasks
+      Rake::Task["db:migrate"].invoke
```

Comment:
This feels a little backwards to me.  What about updating `ActiveRecord::Tasks::DatabaseTasks#migrate` to handle multiple databases (or adding a new method, if necessary)?  Then the `db:migrate` Rake task can be refactored to delegate to that method.

Reply:
I tried out Rails::Command.invoke("db:migrate") but kept getting Don't know how to build task 'rails --task' (See the list of available tasks with rails --tasks) when running the railties test suite. 

`Rails::Command.invoke("db:migrate")` works perfectly in a Rails app but the `railties` test suite has a different behavior which I can't seem to figure out.